### DISCUSSION TITLE
Does "Energy Yield #1" exclude imported energy from the grid?

### DISCUSSION THREAD

**Discussion by StephanRoemer** [JensenNick/huawei_solar_pees#87] [Question / Answer] [answered]
Hey all,

in FusionSolar, my daily PV shows 14 kWh but with the PEES sensor "Energy Yield #1", I get 17 kWh in Home Assistant.
Now I wondered if "Energy Yield #1" includes the imported energy from the grid?

Additionally to charging the battery with the PV, I did boost the battery charging via the grid. Could this be the cause maybe?

**Comment by JensenNick** [comment]
As far as I know "Daily Yield" is the readout from the Huawei Solar PV system. This value is not representative of the actual production because it is being corrected for battery usage and loss in the system. I haven't bothered to try to understand why, when and how, because it seems like very random corrections to me. Here is an example from my plant the 2nd of November 2024. Look at the how the production according to the Huawei data apparently comes to a complete stop although there clearly is solar PV power. It clearly also includes discharging of the battery which is apparent as the value of the sensor increases before sunrise - I've included my readings from august. 
![2024-11-04_09-22-45](https://github.com/user-attachments/assets/3721593a-4c7c-4ba7-9dd8-3a6147bf2fd8)
![2024-11-04_09-28-55](https://github.com/user-attachments/assets/4281f934-867e-45ae-ac7d-7d95eeab2bb0)

**Comment by StephanRoemer** [comment]
Hey JensenNick!

Thanks for your reply! So, I conclude, something is fishy with the Huawei sensor "Inverter Daily Yield".

However, if you check what the FusionSolar App / Website shows, then no hiccups here (no battery added or anything else):
![FusionSolar](https://github.com/user-attachments/assets/f7938ad3-6031-40a8-8c7c-7cc376eb941d)

And now compared to the "Inverter Input Power" (automatically renamed to "Inverter Eingansleistung" by the Huawei integration):
<img width="1349" alt="Bildschirmfoto 2024-11-04 um 21 05 59" src="https://github.com/user-attachments/assets/8a01a76b-003d-4677-969c-e83164856771">


To me, the diagrams look very similar. Meaning, with that sensor we should have the original FusionSolar data from the app / website.
However, the sensor does only return a momentary value and not a daily value.
But I guess this can easily be achieved by creating a utility meter, right?

**Reply by JensenNick** [reply]
The graphs you are sharing are showing power not energy. You need to understand the difference. The Daily Yield from Huawei is energy and can easily be manipulated although the power-graphs are "correct".

**Comment by StephanRoemer** [comment]
Absolutely, I know what you are saying, but what I mean is: 
we can use the data from "inverter input power" and feed it into a Utility Meter to create the daily yield in kWh.

**Reply by JensenNick** [reply, marked as answer]
Ahh... no need to ad any utility meters. The PEES package includes all the sensors and utility meters you need. Take a look here [Wiki Pages](https://github.com/JensenNick/huawei_solar_pees/wiki).

**Reply by StephanRoemer** [reply]
Ah you already created those! D'oh 😄 
Thanks for the heads up!
